spec/features/trainees/filtering_trainees_spec.rb

```diff
@@ -0,0 +1,133 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+RSpec.feature "Filtering trainees" do
+  let!(:assessment_only_trainee) { create :trainee, record_type: "assessment_only" }
```

Comment:
Personally, I've had issues in the past with `let!`  leaking test data across scenarios and not being cleaned up properly in a full test run. Can we move these creates into a method to be called in the scenario? `given_trainees_exist_in_the_system` or something along those lines. I think it'll help make clear which scenarios require which factories